includes/class-wc-subscriptions-order.php

```diff
@@ -622,36 +622,43 @@ public static function get_users_subscription_orders( $user_id = 0 ) {
 		}
 
 		// Get all the customers orders which are not subscription renewal orders
-		$order_ids = get_posts( array(
-			'posts_per_page' => 1,
-			'post_type'      => 'shop_order',
-			'post_status'    => 'any',
-			'fields'         => 'ids',
-			'orderby'        => 'date',
-			'order'          => 'DESC',
-			'meta_query'     => array(
-				array(
-					'key'     => '_customer_user',
-					'compare' => '=',
-					'value'   => $user_id,
-					'type'    => 'numeric',
-				),
-				array(
+		$custom_query_var_handler = function( $query, $query_vars ) {
+			if ( ! empty( $query_vars['_non_subscription_renewal'] ) ) {
+				$query['meta_query'][] = array(
 					'key'     => '_subscription_renewal',
 					'compare' => 'NOT EXISTS',
-				),
-			),
-		) );
+				);
+				unset( $query_vars['_non_subscription_renewal'] );
+			}
+
+			return $query;
+		};
+		add_filter( 'woocommerce_order_data_store_cpt_get_orders_query', $custom_query_var_handler, 10, 2 );
 
-		foreach ( $order_ids as $index => $order_id ) {
-			if ( ! wcs_order_contains_subscription( $order_id, 'parent' ) ) {
-				unset( $order_ids[ $index ] );
+		$all_possible_statuses = array_values( array_unique( array_keys( wc_get_order_statuses() ) ) );
+
+		$args = array(
```

Comment:
In these args, can we add `'return' => 'ids',` since we don't need order objects to be returned?

Reply:
Thank you for pointing that out, @mattallan! Addressed in [cec2cb9](https://github.com/Automattic/woocommerce-subscriptions-core/pull/161/commits/cec2cb9acfbe7234b2fa760b87700c85c2836fe7).